devito/equation.py

```diff
@@ -23,14 +24,18 @@ class Eq(sympy.Eq):
     ----------
     lhs : Function or SparseFunction
         The left-hand side.
-    rhs : expr-like
-        The right-hand side.
+    rhs : expr-like, optional
+        The right-hand side. Defaults to 0.
     subdomain : SubDomain, optional
         To restrict the computation of the Eq to a particular sub-region in the
         computational domain.
     coefficients : Substitutions, optional
         Can be used to replace symbolic finite difference weights with user
         defined weights.
+    implicit_dims : Dimension or list of Dimension, optional
```

Comment:
How 'exposed' do we want `implicit_dims` to actually be? If we want it in the docstring I think we should include a few clear examples (which I could add next week)? 
(Currently it takes a little bit of playing around to actually work out what this functionality does).

Reply:
Agreed. In fact, this might not be exposed to the user level at all. 

If you look at the commit history, you see at some point I had introduced a Step class, subclass of Eq, which was basically an Eq with this additional parameter. Step wouldn't have been exposed in the user API. However, I then quickly realised that then I'd have needed a StepInc class as well -- you can see the pattern here. It was horrible. So the shortcut was to do add it directly to Eq/Inc. 

I can't see how ``implicit_dims`` should be helpful to the average user. So I'm happy to drop it from the docstring ("partial hiding"); any other suggestions would be more than welcome